File: Math/2894. Divisible and Non-divisible Sums Difference/soln.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function (n, m) {
  let totSum = (n * (n + 1)) / 2;
  let divSum = sOD(n, m);
  let nonSum = totSum - divSum;
  return nonSum - divSum;
};

function sOD(n, m) {
  let cnt = Math.floor(n / m);
  return (cnt * (cnt + 1) * m) / 2;
}

File: Math/2894. Divisible and Non-divisible Sums Difference/soln.go
func differenceOfSums(n int, m int) int {
    totSum := n * (n + 1) / 2
    divSum := sOD(n, m)
    nonSum := totSum - divSum
    return nonSum - divSum
}

func sOD(n int, m int) int {
    cnt := int(math.Floor(float64(n) / float64(m)))
    return m * cnt * (cnt + 1) / 2
}